model/monitorCache/implementation/SimpleMonitoringStorage.php

```diff
@@ -0,0 +1,734 @@
+        $groupColumns = [];
+        $defaultOptions = [
+            'order' => join(' ', [static::DEFAULT_SORT_COLUMN, static::DEFAULT_SORT_ORDER, static::DEFAULT_SORT_TYPE]),
+            'offset' => 0,
+            'asArray' => false
+        ];
+        $options = array_merge($defaultOptions, $options);
+
+        $options['order'] = $this->prepareOrderStmt($options['order']);
+        $fromClause = "FROM " . self::TABLE_NAME . " t ";
+
+        $whereClause = $this->prepareCondition($criteria, $this->queryParams, $selectClause);
+        if ($whereClause !== '') {
+            $whereClause = 'WHERE ' . $whereClause;
+        }
+
+        $selectClause = "SELECT " . implode(','.PHP_EOL, $selectColumns) . PHP_EOL;
+
+        $sql = $selectClause . ' ' . $fromClause . PHP_EOL .
+            implode(PHP_EOL, $this->joins) . PHP_EOL .
+            $whereClause . PHP_EOL;
+
+        if (!empty($groupColumns)) {
+            if (!in_array('t.delivery_execution_id', $groupColumns)) {
+                $groupColumns[] = 't.delivery_execution_id';
+            }
+            $sql .= 'GROUP BY ' . implode(',', $groupColumns) . PHP_EOL;
+        }
+
+        $sql .= "ORDER BY " . $options['order'];
+
+        if (isset($options['limit']))  {
+            $sql = $this->getPersistence()->getPlatForm()->limitStatement($sql, $options['limit'], $options['offset']);
+        }
+
+        $stmt = $this->getPersistence()->query($sql, $this->queryParams);
+
+        $data = $stmt->fetchAll(PDO::FETCH_ASSOC);
+
+        if ($options['asArray']) {
+            $result = $data;
+        } else {
+            foreach($data as $row) {
+                $extraData = [];
+                if (isset($row['extra_data'])) {
+                    $decodedExtraData = json_decode($row['extra_data']);
+                    if (json_last_error() === JSON_ERROR_NONE) {
+                        $extraData = $decodedExtraData;
+                    }
+                }
+                $row['extra_data'] = $extraData;
+
+                $deliveryExecution = ServiceProxy::singleton()->getDeliveryExecution($row[self::COLUMN_DELIVERY_EXECUTION_ID]);
+                $result[] = $this->createMonitoringData($deliveryExecution, $row);
+            }
+        }
+
+        return $result;
+    }
+
+    public function count(array $criteria = []): int
+    {
+        $this->joins = [];
+        $this->queryParams = [];
+
+        $selectClause = "select COUNT(*) FROM (SELECT t.delivery_execution_id ";
+        $fromClause = "FROM " . self::TABLE_NAME . " t ";
+        $whereClause = $this->prepareCondition($criteria, $this->queryParams, $selectClause);
+        if ($whereClause !== '') {
+            $whereClause = 'WHERE ' . $whereClause;
+        }
+        $sql = $selectClause . $fromClause . PHP_EOL .
+            implode(PHP_EOL, $this->joins) . PHP_EOL .
+            $whereClause . PHP_EOL .
+            'GROUP BY t.' . self::DELIVERY_EXECUTION_ID . ') as count_q';
```

Comment:
The same, I think we don't need a `GROUP BY` statement any more, but worth to check. And probably with just one table in the query we can get count without nested SELECT.

Reply:
sure, fixed